File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit=0;
        int min=prices[0];
        int n=prices.size();
        for (int i=1;i<n;i++){
            if(prices[i]<min) min=prices[i];
            else if(prices[i]-min>profit) profit=prices[i]-min;
        }
        return profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit=0;
        int min=prices[0];
        for (int i=1;i<prices.length;i++){
            if(min>prices[i]) min=prices[i];
            int cur=prices[i]-min;
            if(cur>profit) profit=cur;
        }
        return profit;
    }
}